src/Chapter3.hs

```diff
@@ -971,6 +1114,22 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Sunday | Monday | Tuesday | Wednesday | Thursday | Friday | Saturday
+            deriving (Show, Eq, Ord, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend day = day == minBound || day == maxBound
+
+nextDay :: Weekday -> Weekday
+nextDay day = if day == maxBound
+              then minBound
+              else succ day
+
+daysToParty :: Weekday -> Int
+daysToParty day = if day > Friday
+                  then (fromEnum day - fromEnum Friday) + fromEnum Friday
+                  else fromEnum Friday - fromEnum day
```

Comment:
That is right!
You can actually simplify it a bit. If you use the fact that there are 7 days in the week, and the way how `mod` works on negatives, then it could be done easier:

```haskell
>>> mod (-1) 7
6
```